path_trie.go

```diff
@@ -14,10 +14,29 @@ type PathTrie struct {
 	children  map[string]*PathTrie
 }
 
+// PathTrieConfig for building a path trie with different segmenter
+type PathTrieConfig struct {
+	segmenter StringSegmenter
+}
+
 // NewPathTrie allocates and returns a new *PathTrie.
-func NewPathTrie() *PathTrie {
+func NewPathTrie(configs ...PathTrieConfig) *PathTrie {
+	var segmenter StringSegmenter
+	for _, config := range configs {
```

Comment:
I don't see a reason this is taking a slice of multiple configs. A typical constructor pattern accepts a *Config of some sort.

```
func NewPathTrie(config *PathTrieConfig) *PathTrie {
  segmenter := config.Segmenter
  if segmenter == nil {
    segmenter = PathSegmenter
  }
  return &PathTrie{
    segmenter: segmenter
  }
}
```

Reply:
I introduced var args to avoid of touching too much code. I have revised the impl by introducing a new constructor which accept the PathTrieConfig. Pls. take a look.